### DISCUSSION TITLE
Adding the ability to stop polling

### DISCUSSION THREAD

**Discussion by ahmedash95** [livewire/livewire#1071] [4. General]
Right now in the livewire, it's possible to set polling to update the component every x seconds. 

AND it would be awesome to have the ability to stop it at some point. what I've in my mind that can make this possible and easy is to use the return type of a method.

so in livewire blade file 
```blade
<div wire:poll.1s="progress">
    In progress {{ '%' . $percent }}
</div>
```

and in livewire component class

```php
class Test extends Component
{
    public $batchId;

    public $percent;

    public function progress() {
        $this->percent = $percent = Batch::findById($this->batchId)->progress();
        if ($percent == 100) {
            return false;
        }
    }
}
```

When `progress` method returns false livewire would stop polling and that would prevent send unnecessary requests

**Comment by winterant** [comment]
Does it solved in livewire v3.x?

**Reply by PhiloNL** [reply]
Yes, I think this also works in v3 by simply removing the poll attribute to stop polling:

```
<div @if($progress !== 100) wire:poll.1s="progress" @endif>
    In progress {{ '%' . $percent }}
</div>
```
